Resolver traverse levels

oatkiller opened the pull request:
## Summary

Recreated https://github.com/elastic/kibana/pull/71777 due to some sort of github issue

* `IndexedProcessTree` now owns the concern of defining the order of siblings
* `IsometricTaxiLayout` now owns the concept of `ariaLevels`
* added `datetime` method to `process_event` model which returns a time in ms since unix epoch for the event
* renamed some resolver selectors
* added resolver selector: `ariaLevel`
* added 'data' selector: `followingSibling` (used for aria-flowto)
* added resolver selector `ariaFlowtoNodeID` which takes a nodeID, and returns its following sibling's node id (if that sibling is visible.) By only returning visible siblings, we ensure that `aria-flowto` will point to an html ID that is in the dom.


![image](https://user-images.githubusercontent.com/35559/87480049-90335300-c5fa-11ea-9f40-5fb6f17e93bc.png)

### Checklist

- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

michaelolo24 commented on x-pack/plugins/security_solution/public/resolver/store/selectors.ts:
```diff
@@ -230,15 +240,50 @@ export const relatedEventTotalForProcess = composeSelectors(
+  visibleNodesAndEdgeLines,
+  composeSelectors(dataStateSelector, dataSelectors.followingSibling),
+  (visibleNodesAndEdgeLinesAtTime, followingSibling) => {
+    return defaultMemoize((time: number) => {
```
clever

oatkiller replied:
been thinking about taking advantage of reselect's parametric selectors. siem uses em. /shrug

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61963/)
* Commit: ab858201c8ab031339bf12f073850b4184f8e990

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/ab858201c8ab031339bf12f073850b4184f8e990) | diff | [baseline](https://github.com/elastic/kibana/commit/99255d824d17125fbb821feee663e6b01d8a0009) |
| --- | --- | --- | --- |
| securitySolution | 9.3MB | +1.7KB | 9.3MB |

#### miscellaneous assets size
| id | [value](https://github.com/elastic/kibana/commit/ab858201c8ab031339bf12f073850b4184f8e990) | diff | [baseline](https://github.com/elastic/kibana/commit/99255d824d17125fbb821feee663e6b01d8a0009) |
| --- | --- | --- | --- |
| upgradeAssistant | 22.5KB | -5.0B | 22.5KB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61963/","number":"61963","commit":"ab858201c8ab031339bf12f073850b4184f8e990"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61970/)
* Commit: 7b5b69ceaf2a9ba30620901357c3c9d4dd08c87f

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/7b5b69ceaf2a9ba30620901357c3c9d4dd08c87f) | diff | [baseline](https://github.com/elastic/kibana/commit/99255d824d17125fbb821feee663e6b01d8a0009) |
| --- | --- | --- | --- |
| securitySolution | 9.3MB | +1.9KB | 9.3MB |

#### miscellaneous assets size
| id | [value](https://github.com/elastic/kibana/commit/7b5b69ceaf2a9ba30620901357c3c9d4dd08c87f) | diff | [baseline](https://github.com/elastic/kibana/commit/99255d824d17125fbb821feee663e6b01d8a0009) |
| --- | --- | --- | --- |
| upgradeAssistant | 22.6KB | +15.0B | 22.5KB |

### History
* :green_heart: [Build #61963](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61963/) succeeded ab858201c8ab031339bf12f073850b4184f8e990

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61963/","number":"61963","commit":"ab858201c8ab031339bf12f073850b4184f8e990"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61970/","number":"61970","commit":"7b5b69ceaf2a9ba30620901357c3c9d4dd08c87f"}]}
PIPELINE-->

oatkiller commented on x-pack/plugins/security_solution/server/endpoint/routes/resolver/utils/pagination.ts:
```diff
@@ -61,7 +61,7 @@ export class PaginationBuilder {
     const cursor = {
       timestamp: lastResult['@timestamp'],
-      eventID: eventId(lastResult),
+      eventID: eventId(lastResult) === undefined ? '' : String(eventId(lastResult)),
```
@jonathan-buttner if this is cool, can i get a re-thumb? ty

oatkiller commented:
made sure it still works after addressing comments 
![image](https://user-images.githubusercontent.com/35559/87589129-c7fdd180-c6b2-11ea-8ea9-cb3e4d42cf4a.png)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/62031/)
* Commit: 3effa8b5f6f97aad8a4d8e0ba4cb2394e82b9bd7

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/3effa8b5f6f97aad8a4d8e0ba4cb2394e82b9bd7) | diff | [baseline](https://github.com/elastic/kibana/commit/99255d824d17125fbb821feee663e6b01d8a0009) |
| --- | --- | --- | --- |
| securitySolution | 9.3MB | +1.9KB | 9.3MB |

### History
* :green_heart: [Build #61970](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61970/) succeeded 7b5b69ceaf2a9ba30620901357c3c9d4dd08c87f
* :green_heart: [Build #61963](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61963/) succeeded ab858201c8ab031339bf12f073850b4184f8e990

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61963/","number":"61963","commit":"ab858201c8ab031339bf12f073850b4184f8e990"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/61970/","number":"61970","commit":"7b5b69ceaf2a9ba30620901357c3c9d4dd08c87f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/62031/","number":"62031","commit":"3effa8b5f6f97aad8a4d8e0ba4cb2394e82b9bd7"}]}
PIPELINE-->
